src/equations/polytropic_euler_2d.jl

```diff
@@ -0,0 +1,255 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+@doc raw"""
+    PolytropicEulerEquations2D(gamma, kappa)
+
+The polytropic Euler equations
+```math
+\partial t
+\begin{pmatrix}
+\rho \\ \rho v_1 \\ \rho v_2
+\end{pmatrix}
++
+\partial x
+\begin{pmatrix}
+ \rho v_1 \\ \rho v_1^2 + \kappa\rho^\gamma \\ \rho v_1 v_2
+\end{pmatrix}
++
+\partial y
+\begin{pmatrix}
+\rho v_2 \\ \rho v_1 v_2 \\ \rho v_2^2 + \kappa\rho^\gamma
+\end{pmatrix}
+=
+\begin{pmatrix}
+0 \\ 0 \\ 0
+\end{pmatrix}
+```
+for an ideal gas with ratio of specific heats `gamma`
+in two space dimensions.
+Here, ``\rho`` is the density and ``v_1`` and`v_2` the velocities and
+```math
+p = \kappa\rho^\gamma
+```
+the pressure, which we replaced using this relation.
+
+"""
+struct PolytropicEulerEquations2D{RealT <: Real} <:
+       AbstractPolytropicEulerEquations{2, 3}
+    gamma::RealT               # ratio of specific heats
+    kappa::RealT               # fluid scaling factor
+
+    function PolytropicEulerEquations2D(gamma, kappa)
+        gamma_, kappa_ = promote(gamma, kappa)
+        new{typeof(gamma_)}(gamma_, kappa_)
+    end
+end
+
+function varnames(::typeof(cons2cons), ::PolytropicEulerEquations2D)
+    ("rho", "rho_v1", "rho_v2")
+end
+varnames(::typeof(cons2prim), ::PolytropicEulerEquations2D) = ("rho", "v1", "v2")
+
+"""
+initial_condition_convergence_test(x, t, equations::PolytropicEulerEquations2D)
+
+Manufactured smooth initial condition used for convergence tests.
+"""
+function initial_condition_convergence_test(x, t, equations::PolytropicEulerEquations2D)
+    # manufactured initial condition from Winters (2019) [0.1007/s10543-019-00789-w]
+    # domain must be set to [0, 1] x [0, 1]
+    h = 8 + cos(2 * pi * x[1]) * sin(2 * pi * x[2]) * cos(2 * pi * t)
+
+    return SVector(h, h / 2, 3 * h / 2)
+end
+
+"""
+source_terms_eoc_test_polytropic(u, x, t, equations::PolytropicEulerEquations2D)
+
```

Comment:
```suggestion
    source_terms_convergence_test(u, x, t, equations::PolytropicEulerEquations2D)

Source terms used for convergence tests in combination with
[`initial_condition_convergence_test`](@ref).
```
Such that the docstring matches the function name.